api/v1/model_types.go

```diff
@@ -20,12 +20,12 @@ type ModelSpec struct {
 	// Resources are the compute resources required by the container.
 	Resources *Resources `json:"resources,omitempty"`
 
-	// BaseModel should be set in order to mount another model to be
+	// Model should be set in order to mount another model to be
 	// used for transfer learning.
-	BaseModel *ObjectRef `json:"baseModel,omitempty"`
+	Model *ObjectRef `json:"model,omitempty"`
 
 	// Dataset to mount for training.
-	TrainingDataset *ObjectRef `json:"trainingDataset,omitempty"`
+	Dataset *ObjectRef `json:"dataset,omitempty"`
```

Comment:
Same comment. I think this is a good rename and support it just calling out that we shouldn't take these lightly.